Orchestration/IntegrationService/Implementations/AzureResourceManagerDeploymentService.ps1

```diff
@@ -364,9 +364,11 @@ Class AzureResourceManagerDeploymentService: IDeploymentService {
                 -and $_.ExpiresOn -gt (Get-Date)) {
```

Comment:
Can you add, $null -ne $_.TenantId?

Also, this seems like even if the value is found, the algorithm will keep looping until it finishes all $cacheItems, can you do, break; or return after the first if condition is met?

Reply:
Updated for null check on TenantId.